### PR TITLE
Add Typedoc and function documentation

### PR DESCRIPTION
**Pull request by mj850**
We want to add TypeDoc to provide documentation for sei-js packages (core and react).

This PR adds the ability to generate a Docs html page using TypeDoc.
Also adds basic placeholder documentation for the exported functions in sei-js/core package.

### CODE REVIEW

File: .github/workflows/deploydocs.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Deploy Docs
+
+on:
+  push:
+    branches:
+      - main
+      - mj/setupTypedoc
```
Comment (besated) [inline .github/workflows/deploydocs.yml:7]: can we remove the workflow run on the `mj/setupTypedoc` branch

File: .github/workflows/deploydocs.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Setup Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
+      - name: Install Dependencies
+        run: yarn
+
+      - name: Create Docs
+        run: yarn build:all
```
Comment (besated) [inline .github/workflows/deploydocs.yml:44]: do we need to build all to create the docs?

File: README.md [hunk 3]
```diff
@@ -6,10 +6,15 @@ SeiJS is a monorepo that contains multiple NPM libraries for writing application
 
 Please check [our documentation](https://docs.seinetwork.io/front-end-development/javascript-tutorial) for notes on how to get up and running. The tutorial has examples on how to connect to a Sei wallet, query an RPC endpoint, transfer tokens, IBC transfer, and interact with contracts.
 
+The `jsexamples.md` file contains examples of how to use the classes and methods in this library to perform interactions with the chain.
```
Comment (besated) [inline README.md:9]: nit: can we call this `EXAMPLES.md`?

File: package.json [hunk 4]
```diff
@@ -7,7 +7,7 @@
 		"packages/*"
 	],
 	"scripts": {
-		"build:all": "nx run-many --target=build",
+		"build:all": "nx run-many --target=build && npx typedoc",
```
Comment (besated) [inline package.json:10]: do we have to build all packages for the typedoc? if not i think we can make this command separate

**Review by besated — CHANGES_REQUESTED** [review]
looks good, thanks for adding all these comments it really helps!

File: package.json (hunk 4 above)
Comment (mj850) [reply package.json]: Nope, just proto -> core, react. Updated with a new docs script that builds what is required for docs and the docs themselves.

**Review by besated — APPROVED** [review]

**Pull request merged**